common/kubernetes/configs.go

```diff
@@ -120,9 +120,6 @@ func GetKubeClientConfig(uri *url.URL) (*kube_client.Config, error) {
 	if len(kubeConfig.Host) == 0 {
 		return nil, fmt.Errorf("invalid kubernetes master url specified")
 	}
-	if len(kubeConfig.GroupVersion.Version) == 0 {
```

Comment:
Sorry I don't have time to investigate why NewNonInteractiveDeferredLoadingClientConfig() returns empty version. On the other hand, the version in the kubeconfig isn't useful currently, it will be given a default value anyway, so I can LGTM the PR. It's just that you might get surprises when we have more groups that have multiple versions and the default version isn't the one you want (currently only `batch` have two versions).